File: workflow/activities/embed_text.py
from temporalio import activity
from ml.embeder import embeder


@activity.defn
async def embed_text(text: str) -> list[float]:
    """
    Activity to generate embeddings for text using OpenAI embeddings.
    
    Args:
        text: The text to embed
        
    Returns:
        A list of floats representing the embedding vector
    """
    embedding = embeder.embed_query(text)
    return embedding


@activity.defn
async def embed_texts(texts: list[str]) -> list[list[float]]:
    """
    Activity to generate embeddings for multiple texts in batch.
    
    Args:
        texts: List of texts to embed
        
    Returns:
        List of embedding vectors
    """
    embeddings = embeder.embed_documents(texts)
    return embeddings

File: ts/workers/runner/src/activities/embed-text.ts
import { openai } from '@ai-sdk/openai';
import { embed, embedMany } from 'ai';
import { getConfig } from '@fdsa/shared';

/**
 * Activity to generate embeddings for a single text using OpenAI embeddings.
 * 
 * @param text - The text to embed
 * @returns A list of floats representing the embedding vector
 */
export async function embedText(text: string): Promise<number[]> {
  const config = getConfig();
  
  const result = await embed({
    model: openai.embedding(config.openai.embeddingModel, {
      dimensions: config.openai.embeddingDimensions,
    }),
    value: text,
  });

  return result.embedding;
}

/**
 * Activity to generate embeddings for multiple texts in batch.
 * 
 * @param texts - List of texts to embed
 * @returns List of embedding vectors
 */
export async function embedTexts(texts: string[]): Promise<number[][]> {
  const config = getConfig();
  
  const result = await embedMany({
    model: openai.embedding(config.openai.embeddingModel, {
      dimensions: config.openai.embeddingDimensions,
    }),
    values: texts,
  });

  return result.embeddings;
}
